roles/openshift_label_nodes/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- name: Get Nodes Config
+  shell: osc get nodes -o json | sed -e '/"resourceVersion"/d'
+  register: output
+
+- name: Set node regions
+  set_fact:
+    node_facts: "{{ output.stdout | from_json
+                  | oo_set_node_label('region', {{oln_node_region}}) }}"
+
+- name: Set infra node labels
+  set_fact:
+    node_final_facts: "{{ node_facts
+                  | oo_set_node_label('region', {{ocom_infra_node_label}}, {{oln_infra_node_filter_key}}, {{oln_infra_node_filter_value}}) }}"
+
+- name: Write node config to temp file
+  copy:
+    content: "{{ node_final_facts }}"
+    dest: /tmp/nodes.json
```

Comment:
Instead of writing to a config file, you can use osc update --patch, which takes a json string and will just update the elements in the json string

It probably wouldn't hurt to go ahead and still get the output of osc get nodes -o json and see if any changes are needed, otherwise we'll report changes every run.